Wikipedia/Code/AppDelegate.m

```diff
@@ -52,15 +51,52 @@ - (BOOL)application:(UIApplication*)application didFinishLaunchingWithOptions:(N
     [vc launchAppInWindow:self.window];
     self.appViewController = vc;
 
+    [self updateDynamicIconShortcutItems];
+    
     NSLog(@"%@", [[[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask] lastObject]);
 
     return YES;
 }
 
+- (void)updateDynamicIconShortcutItems {
+    self.shortcutItemSelectedAtLaunch = nil;
+
+    if (![[UIApplication sharedApplication] respondsToSelector:@selector(shortcutItems)]) return;
+    
+    UIApplicationShortcutItem* (^makeShortcut)(NSString*, NSString*, NSString*, NSString*) = ^(NSString* type, NSString* title, NSString* subtitle, NSString* icon) {
```

Comment:
would you consider moving this into a category—as it's basically a custom/convenience factory/initializer for `UIApplicationShortcutItem`? doesn't seem to gain anything from being a block (AFAICT) and only makes this method longer.